src/modules/haskell.rs

```diff
@@ -0,0 +1,60 @@
+use ansi_term::Color;
+use std::process::Command;
+
+use super::{Context, Module};
+
+/// Creates a module with the current Haskell Stack version
+///
+/// Will display the Haskell version if any of the following criteria are met:
+///     - Current directory contains a `stack.yaml` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_haskell_project = context
+        .try_begin_scan()?
+        .set_files(&["stack.yaml"])
+        .is_match();
+
+    if !is_haskell_project {
+        return None;
+    }
+
+    match get_haskell_version() {
+        Some(haskell_version) => {
+            const HASKELL_CHAR: &str = "λ ";
+
+            let mut module = context.new_module("haskell");
+            let module_style = module
+                .config_value_style("style")
+                .unwrap_or_else(|| Color::Red.bold());
+            module.set_style(module_style);
+
+            let formatted_version = format_haskell_version(&haskell_version)?;
+            module.new_segment("symbol", HASKELL_CHAR);
+            module.new_segment("version", &formatted_version);
+
+            Some(module)
+        }
+        None => None,
+    }
+}
+
+fn get_haskell_version() -> Option<String> {
+    match Command::new("stack")
```

Comment:
I'm not too aware of `stack` and `ghc` but should you maybe check the yaml file first for a set compiler? https://docs.haskellstack.org/en/v1.0.2/yaml_configuration/#compiler Reading from file is usually faster than starting some other command

Reply:
The `compiler` is optional and can't be relied on.

I'm also planning to add support for `package.yaml` and `.cabal` -styled projects, then it'll make sense to first check files for compiler version and if it isn't available - run `stack` or `ghc`.